CriteoPublisherSdk/Sources/Network/CR_NetworkManager.h

```diff
@@ -42,10 +42,16 @@ typedef void (^CR_NMResponse)(NSData *_Nullable data, NSError *_Nullable error);
 
 - (void)getFromUrl:(NSURL *)url responseHandler:(nullable CR_NMResponse)responseHandler;
 
-// Assumes all POST calls are made via JSON
-- (void)postToUrl:(NSURL *)url
-           postBody:(NSDictionary *)postBody
-    responseHandler:(nullable CR_NMResponse)responseHandler;
+/**
+ * Send POST request with JSON body
+ * @param url Request URL
+ * @param postBody Dictionary that will be posted as JSON
+ * @param responseHandler Response handler
+ * @return The data posted
```

Comment:
Feels a bit weird to return to the caller a value that it (the caller) has already knowledge of.  I know there is some legacy here and I understand that `postToUrl` is handling serialization, but I wanted to get your perspective on how this could be improved in the future.

Reply:
Clearly this is was the shortest past solution w/o being too intrusive / refactoring many things.
If we had time, I would rewrite most of the network stack of the SDK as the current Caller -> ApiHandler / NetworkManager could deserve some improvements (there are some tickets in the project backlog about this)
A decent middle ground solution could be to add extra arguments to this method to specify that we want to log JSON posted data, as the problem is that we do not want to log all queries, but we want JSON format (TS spec) which is only available there.

Reply:
Yes, it does. 

> A decent middle ground solution could be to add extra arguments to this method to specify that we want to log JSON posted data, as the problem is that we do not want to log all queries, but we want JSON format (TS spec) which is only available there.

It does prevent the need of returning the serialized JSON, however it doesn't allow customizing the log message relative to the type of request being made.

I wonder whether `postBody` can be used for logging?

Reply:
The idea was to have one postBody argument to be used in the log message to address this.

Anyway, for now, don't you think the current simple approach is clean enough until a day where some refactoring can be done in this part of the codebase?

Reply:
Sure, lgtm.